### ISSUE TITLE
Test failure with Python 3.13

### ISSUE THREAD

**Issue by opoplawski** [gevent/gevent#2037]
* gevent version: 24.2.1 - source
* Python version: 3.13b2
* Operating System: Fedora Rawhide x86_64

### Description:

Fedora has update rawhide to python 3.13.  gevent tests fail with:
```python-traceback
    Traceback (most recent call last):
      File "<frozen runpy>", line 198, in _run_module_as_main
      File "<frozen runpy>", line 88, in _run_code
      File "/builddir/build/BUILD/python-gevent-24.2.1-build/BUILDROOT/usr/lib64/python3.13/site-packages/gevent/monkey.py", line 1358, in <module>
        main()
        ~~~~^^
      File "/builddir/build/BUILD/python-gevent-24.2.1-build/BUILDROOT/usr/lib64/python3.13/site-packages/gevent/monkey.py", line 1310, in main
        patch_all(**args)
        ~~~~~~~~~^^^^^^^^
      File "/builddir/build/BUILD/python-gevent-24.2.1-build/BUILDROOT/usr/lib64/python3.13/site-packages/gevent/monkey.py", line 1219, in patch_all
        patch_thread(Event=Event, _warnings=_warnings)
        ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/builddir/build/BUILD/python-gevent-24.2.1-build/BUILDROOT/usr/lib64/python3.13/site-packages/gevent/monkey.py", line 187, in ignores
        return func(*args, **kwargs)
      File "/builddir/build/BUILD/python-gevent-24.2.1-build/BUILDROOT/usr/lib64/python3.13/site-packages/gevent/monkey.py", line 826, in patch_thread
        main_thread.__real_tstate_lock = main_thread._tstate_lock
                                         ^^^^^^^^^^^^^^^^^^^^^^^^
    AttributeError: '_MainThread' object has no attribute '_tstate_lock'
```

**Comment by rominf** [comment]
This is caused by changes in CPython: https://github.com/python/cpython/pull/114839.

> The combination of _tstate_lock and _ThreadHandle is replaced by _ThreadHandle and the supporting infrastructure for _tstate_lock is removed.

Also, the latest Cython does not fully support Python 3.13: https://github.com/cython/cython/issues/6251 (some tests fail on my machine as well).

I suggest waiting for a Cython release that fully supports Python 3.13 before attempting to fix this issue.

**Comment by ngoldbaum** [comment]
There's a Cython nightly wheel if you want to get started on this now: https://anaconda.org/scientific-python-nightly-wheels/cython

**Comment by opoplawski** [comment]
Can this get addressed?  Fedora 41 is about to be released with Python 3.13 and this is causing issues with a number of packages.  Thanks.
